src/test/run-make/issue-26006/Makefile

```diff
@@ -0,0 +1,16 @@
+-include ../tools.mk
+
+ifndef IS_WINDOWS
+all: time
+
+time: libc
+	mkdir -p out/time out/time/deps
+	ln -sf out/libc/liblibc.rlib out/time/deps/
+	$(RUSTC) in/time/lib.rs -Ldependency=out/time/deps/
```

Comment:
As far as I understand -Ldependency= adds some path to a list of directories in which to look for library files, right? Then this fails to use symlinked libc and proceeds to find one in rustc’s own distribution, which is why this doesn’t fail.

If this specified --extern libc=, though, then it would be a [different problem](https://github.com/rust-lang/rust/issues/26043).